packages/lesswrong/lib/routeUtil.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { useContext } from 'react';
+import qs from 'qs';
+import { NavigationContext, LocationContext, SubscribeLocationContext } from 'meteor/vulcan:core';
+
+// Given the props of a component which has withRouter, return the parsed query
+// from the URL.
+export function parseQuery(location) {
+  let query = location?.search;
+  if (!query) return {};
+  
+  // The unparsed query string looks like ?foo=bar&numericOption=5&flag but the
+  // 'qs' parser wants it without the leading question mark, so strip the
+  // question mark.
+  if (query.startsWith('?'))
+    query = query.substr(1);
+    
+  return qs.parse(query);
+}
+
+// React Hook which returns the page location (parsed URL and route).
```

Comment:
Maybe a brief comment here about when you want to use this hook vs the withLocation HoC below? Like, kinda explaining how to use react hooks, but also might be helpful for new devs.